### PR TITLE
less: 563 -> 581.2

### PR DESCRIPTION
**Pull request by kidonng**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/less/default.nix [hunk 1]
```diff
@@ -2,11 +2,11 @@
...
-  version = "563";
+  version = "581.2";
 
   src = fetchurl {
     url = "http://www.greenwoodsoftware.com/${pname}/${pname}-${version}.tar.gz";
-    sha256 = "16lsvk88vwjwp5ax1wnll44wxwnzs8lb2fn90xx2si64kwmnsnyf";
+    sha256 = "0fyqslvrasv19qjvqrwfwz2n7mnm93y61x9bcx09ga90mxyb8d6f";
   };
 
   configureFlags = [ "--sysconfdir=/etc" ] # Look for ‘sysless’ in /etc.
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/less/default.nix:12]: Can we update the homepage to https://www.greenwoodsoftware.com/less/ ?

**Comment by SuperSandro2000** [comment]
changelog 

```
Less version 581
Version 581 was released on 7 Apr 2021. An upgraded version 581.2 was released on 28 Apr 2021.

These are the differences between version 563 and version 581:

Change ESC-u command to toggle, not disable, highlighting per man page.
Add ESC-U command.
Add ctrl-W search modifier for wrapping search.
F command can be interrupted by ^X.
Support OSC 8 hyperlinks when -R is in effect.
g command with no number will ignore -j and put first line at top of screen.
Multiple + or -p command line options are handled better.
Add the --incsearch option.
Add the --line-num-width option.
Add the --status-col-width option.
Add the --use-color and --color options.
Display -w highlight even if highlighted line is empty.
If search result is in a long line, scroll to ensure it is visible.
Editing the same file under different names now creates only one entry in the file list.
Make visual bell more visible on some terminals.
Ring end-of-file bell no more than once per second.
Build can use either Python or Perl for Makefile.aut operations.
Fix crash when using the @ search modifier.
Fix crash in the 's' command due to duplicate free.
Fix realpath crash on Darwin.
```

**Review by SuperSandro2000 — APPROVED** [review]
seems to work fine for me

File: pkgs/tools/misc/less/default.nix (hunk 1 above)
Comment (happysalada) [reply pkgs/tools/misc/less/default.nix:12]: Just to clarify here, it's a matter of switching http -> https

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 127289` at a44dc820 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>ovftool</li>
    <li>spark</li>
  </ul>
</details>
<details>
  <summary>240 packages skipped due to time constraints:</summary>
  <ul>
    <li>R</li>
    <li>Sylk</li>
    <li>altair</li>
    <li>android-studio (androidStudioPackages.stable)</li>
    <li>androidStudioPackages.beta</li>
    <li>androidStudioPackages.canary</li>
    <li>androidStudioPackages.dev</li>
    <li>anki-bin</li>
    <li>appimage-run</li>
    <li>apple-music-electron</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>18 packages built successfully:</summary>
  <ul>
    <li>awscli</li>
    <li>bash-my-aws</li>
    <li>flutter (flutterPackages.stable)</li>
    <li>git-recent</li>
    <li>git-remote-codecommit</li>
    <li>gitRepo</li>
    <li>home-manager</li>
    <li>iceshelf</li>
    <li>less</li>
    <li>nimbo</li>
    <li>patroni</li>
    <li>powershell</li>
    <li>ydiff (python38Packages.ydiff)</li>
    <li>python39Packages.ydiff</li>
    <li>ranger</li>
    <li>ssm-session-manager-plugin</li>
    <li>steam</li>
    <li>unison-ucm</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/tools/misc/less/default.nix:21:5:
```
   |
21 |     license = licenses.gpl3;
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/tools/misc/less/default.nix (hunk 1 above)
Comment (kidonng) [reply pkgs/tools/misc/less/default.nix:12]: > Can we update the homepage to [greenwoodsoftware.com/less](https://www.greenwoodsoftware.com/less/) ?

Sure, why not

> Just to clarify here, it's a matter of switching http -> https

Thanks for clarifying!

**Pull request merged**